docs/reading_files.rst

```diff
@@ -268,3 +268,34 @@ You only need these details if you think caching may be causing problems.
 
 JSON was chosen as it can be easily inspected manually, and it's reasonably
 efficient to load the entire file.
+
+File Reading Issues
+-------------------
+
+Files at European XFEL storage migrate over time from GPFS to PNFS. The data 
+on PNFS usually always available for reading. But sometimes, reading may require
+staging from the tape to dCache. If there is staging queue, the operation can take
+an indefinitely long time (days or even weeks) and any IO operations will be 
+blocked for this time.
+
+To determine the files which require staging or are lost, use the script::
+
+    extra-date-locality <run directory>
+
+It returns a list of files which are currently located only on slow media for some
+reasons and, separetelly, which are lost at all.
+
+If the files are not essential for analysis, then they can be filtered out using 
+filter :func:`ls_ondisk` from :mod:`extra_data.locality`::
+
+    from extra_data.locality import lc_ondisk
+    run = open_run(proposal=700000, run=1, file_filter=lc_ondisk)
+
+``file_filter`` must be a callable which takes a list as a single argument and
+returns filtered list.
+
+**Note: the reading of file locality on PNFS very expensive operation.
```

Comment:
```suggestion
**Note: Reading the file locality on PNFS is an expensive operation.
```

'Very' seems misplaced - it's still cheap compared to loading data from tape, for instance. :-)